kuma/health/middleware.py

```diff
@@ -0,0 +1,53 @@
+from django.db import DatabaseError
+from django.http import HttpResponse
+from django.views.decorators.http import require_GET
+
+from kuma.wiki.models import Document
+
+
+class HealthCheckMiddleware(object):
+    """
+    This middleware implements the "liveness" and "readiness" endpoints
+    used to manage an application within a Kubernetes cluster.
+    """
+
+    def process_request(self, request):
+        if request.path == '/healthz':
+            return handle_liveness(request)
+
+        if request.path == '/readiness':
+            return handle_readiness(request)
+
+
+@require_GET
```

Comment:
I think ``HEAD`` should be allowed for  the liveness endpoint, especially since the response has no content.  The [require_safe](https://docs.djangoproject.com/en/1.8/topics/http/decorators/#django.views.decorators.http.require_safe) decorator would do that.